core/src/main/java/com/amplifyframework/core/model/query/predicate/QueryPredicateGroup.java

```diff
@@ -100,7 +101,49 @@ public QueryPredicateGroup or(QueryPredicate predicate) {
      * @return a group negating the given group of operations
      */
     public static QueryPredicateGroup not(QueryPredicateGroup predicate) {
-        return new QueryPredicateGroup(Type.NOT, Arrays.asList(predicate));
+        return new QueryPredicateGroup(Type.NOT, Collections.singletonList(predicate));
+    }
+
+    /**
+     * Evaluate the combination of operations associated with
+     * this group of predicates.
+     * @param object The object to evaluate against
+     * @return Evaluated result of this logical combination
+     * @throws IllegalArgumentException when the object contains
+     *          a field with data type that cannot be evaluated
+     * @throws IllegalStateException when the predicate group does
+     *          not contain any predicate element
+     */
+    @Override
+    public boolean evaluate(Object object) throws
+            IllegalArgumentException,
+            IllegalStateException {
+        if (predicates.isEmpty()) {
+            throw new IllegalStateException("A predicate group " +
+                    "must contain at least one predicate element");
+        }
```

Comment:
If this is true, can you require this in the constructor, instead of waiting until the `evaluate(...)`, here?

Alternately, if you only need >=1 predicates in a specific scenario, can you define that specific scenario in this error message?

Lastly, should this throw a `AmplifyException`, instead? (Not sure)?

Reply:
I decided to make it throw a runtime exception because it figured predicate evaluation is a simple boolean logic operation that shouldn't be try-catched every time.

Also I agree that constructor should guarantee non-empty list of `predicates` instead of error-checking here. Will be fixed.